README.md

```diff
@@ -51,11 +51,13 @@ $shopifyClient = new ShopifyClient([
 ```
 
 Most of the time, you will need to alter the client once you have retrieved the shop info. To that extent, the client offers
-two public methods:
+a public method that allows to create a new client:
 
 ```php
-$shopifyClient->setShopDomain('merchant.myshopify.com');
-$shopifyClient->setAccessToken('your_access_token');
+$newClient = $shopifyClient->withConnectionSettings([
```

Comment:
We are doing that for instance in the Common: https://github.com/maestrooo/shopify-apps-common/blob/master/src/OAuth/ShopifyAuthorizationService.php#L89

Actually speaking about that I think that I'll need to remove the "validateSettings". In our case for instance we initiate the client, but we don't know the access token yet until we have loaded the info from the database.

What about removing completely the "withConnectionSettings", and asking always for a completetely created client? In our code, we may get ride of the the "shopifyClientPrototype" and just create the client using a new (would it be that bad ? Except it would be harder to test).